omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java

```diff
@@ -243,11 +249,41 @@ protected PageableResult doSearch(RequestContext context) {
 				filterByType(orders, context.getType());
 			}
 			
-			return new NeedsPaging<Order>(orders, context);
+			if (StringUtils.isNotBlank(sortParam)) {
+				List<Order> sortedOrder = sortOrdersBasedOnDateActivatedOrDateStopped(orders, sortParam, status);
+				return new NeedsPaging<Order>(sortedOrder, context);
+			}
+			else {
+				return new NeedsPaging<Order>(orders, context);
+			}
 		}
 		
 		return new EmptySearchResult();
 	}
+
+	private static Date getUsableDate(Order order) {
+		return order.getDateStopped() != null ? order.getDateStopped() : order.getAutoExpireDate();
+	}
+	
+	public List<Order> sortOrdersBasedOnDateActivatedOrDateStopped(List<Order> orders, final String sortOrder,
+	        final String status) {
+		List<Order> sortedList = new ArrayList<Order>(orders);
+		Collections.sort(sortedList, new Comparator<Order>() {
+			
+			@Override
+			public int compare(Order o1, Order o2) {
+				if (status.equalsIgnoreCase("inactive")) {
+					return sortOrder.equalsIgnoreCase("ASC") ?
+					        getUsableDate(o1).compareTo(getUsableDate(o2)) : getUsableDate(o2).compareTo(getUsableDate(o1));
+				}
+				else {
+					return sortOrder.equalsIgnoreCase("asc") ? o1.getDateActivated().compareTo(o2.getDateActivated()) : o2
```

Comment:
We should have a fallback to dateCreated here as well... I believe that this branch of the if/else will be hit if status is set to "active" or "any"  (see https://github.com/openmrs/openmrs-module-webservices.rest/blob/master/omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderUtil.java#L40)  so it's possible that we would get orders here that have dateActivated = null.